modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1763,13 +1766,190 @@ def sort_by(
+            if len(self.axes[0]) == 0 or len(self.axes[1]) == 0:
+                return self.copy()
+            if self.dtypes[columns[0]] == object:
+                # This means we are not sorting numbers, so we need our quantiles to not try
+                # arithmetic on the values.
+                method = "inverted_cdf"
+            else:
+                method = "linear"
+
+            def sample_func(df, A=100, k=0.05, q=0.1):
+                """
+                Sample the given partition.
+
+                This function is applied to each row-axis partition in parallel to select samples
+                from each. It samples using the following algorithm:
+                    * If there are <= 100 rows in the dataframe, we pick quantiles over the entire dataframe.
+                    * If there are 100 < # of rows <= 1900, we pick quantiles over the first 100 rows, plus a sample
+                    of 5% of the remaining rows.
+                    * If there are > 1900 rows, we pick quantiles over a sample of 10% of all of the rows.
+
+                These numbers are a heuristic. They were picked such that the size of the sample
+                in scenario 2 scales well. In other words, we picked A, k, and q such that:
+                    A + (len(df) - A)*k = q*len(df)
+                Where q is the proportion we sample in scenario 3, k is the proportion we sample
+                of remaining rows in scenario 2, and A is the threshold below which we just return
+                the entire dataframe, instead of sampling (scenario 1).
+                q = 0.1 and k = 0.05 were picked such that this formula holds for A = 100.
+                """
+                quantiles = [
+                    i / (NPartitions.get() * 2) for i in range(NPartitions.get() * 2)
+                ]
+                # Heuristic for a "small" df we will compute quantiles over entirety of.
+                if len(df) <= A:
+                    return df, np.quantile(df[columns[0]], quantiles, method=method)
+                # Heuristic for a "medium" df where we will include first 100 (A) rows, and sample
+                # of remaining rows when computing quantiles.
+                if len(df) <= A * (1 - k) / (1 - q):
+                    return df, np.quantile(
+                        np.concatenate(
+                            (
+                                df[columns[0]][:100].values,
+                                df[columns[0]][100:].sample(frac=k),
+                            )
+                        ),
+                        quantiles,
+                        method=method,
+                    )
+                # Heuristic for a "large" df where we will sample 10% (q) of all rows to compute quantiles
+                # over.
+                return df, np.quantile(
+                    df[columns[0]].sample(frac=q), quantiles, method=method
+                )
+
+            def pivot_func(samples):
```

Comment:
Could you please add documentation for the params and return type? I think the doc checker will make you do that anyway once you move this function definition outside of `sort_values`.